### PR TITLE
DB-11412 disable preserveLineEndings if oneLineRecords is set

### PR DESCRIPTION
**Pull request by martinrupp**
# Description

Before the fix, preserveLineEndings didn't use parallel readings in SpliceFileVTI. This is because we use another
code path which would not preserve line endings. This is correct, however If we have oneLineRecords, it doesn't make sense to preserve line endings, since we don't read the line endings into the data (for this, we would need multiline-line varchars). So the case oneLineRecords we can set preserveLineEndings to false, which then allows parallel reads in this case.

So, simple solution is to set preserveLineEndings to false if we have oneLineRecords (in Code).
Workaround for customer is to do this manually with global options.

# How to test
This is a performance improvement, see https://splicemachine.atlassian.net/browse/DB-11412 .

### CODE REVIEW

**Review by arnaud-splice — APPROVED** [review]

File: splice_machine/src/main/java/com/splicemachine/derby/vti/SpliceFileVTI.java [hunk 1]
```diff
@@ -161,7 +161,10 @@ boolean getPreserveLineEndings(SpliceOperation op) throws StandardException {
                 quotedEmptyIsNull = vtiOperation.getQuotedEmptyIsNull();
             }
 
-            boolean preserveLineEndings = getPreserveLineEndings(op);
+            // if we have oneLineRecords, it doesn't make sense to preserve line endings, since
+            // we don't read the line endings into the data (for this, we would need multiline-line varchars)
+            boolean preserveLineEndings = oneLineRecords ? false : getPreserveLineEndings(op);
+
             if (oneLineRecords && !preserveLineEndings
                     && (charset==null || charset.toLowerCase().equals("utf-8"))) {
```
Comment (jyuanca) [inline splice_machine/src/main/java/com/splicemachine/derby/vti/SpliceFileVTI.java:169]: can this be "if (oneLineRecords && && (charset==null || charset.toLowerCase().equals("utf-8")))", ignoring preserveLineEndings ?

**Review by jyuanca — CHANGES_REQUESTED** [review]

File: splice_machine/src/main/java/com/splicemachine/derby/vti/SpliceFileVTI.java (hunk 1 above)
Comment (martinrupp) [reply splice_machine/src/main/java/com/splicemachine/derby/vti/SpliceFileVTI.java:169]: yes, of course

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Review by dgomezferro — APPROVED** [review]

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by jyuanca — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

**Comment by jyuanca** [comment]
Do we also need this for 3.0 and 3.1?

**Pull request merged**